src/uattributes/uattributesvalidator.rs

```diff
@@ -360,193 +250,253 @@ impl UAttributesValidators {
-    ///
-    /// # Arguments
+    /// Verifies that attributes for a message representing an RPC request contain a reply-to-address.
     ///
-    /// * `attributes` - `UAttributes` object to validate.
-    ///
-    /// # Returns
+    /// # Errors
     ///
-    /// Returns a `ValidationResult` that is success or failed with a failure message.
+    /// If the [`UAttributes::source`] property does not contain a valid reply-to-address according to
+    /// [`UriValidator::validate_rpc_response`], an error is returned.
     fn validate_source(&self, attributes: &UAttributes) -> Result<(), UAttributesError> {
         if let Some(source) = attributes.source.as_ref() {
             UriValidator::validate_rpc_response(source)
                 .map_err(|e| UAttributesError::validation_error(e.to_string()))
         } else {
-            Err(UAttributesError::validation_error("Missing Source"))
+            Err(UAttributesError::validation_error("Attributes for a request message must contain a reply-to address in the source property"))
         }
     }
 
-    /// Validates that attributes for a message meant for an RPC request has a destination sink.
-    /// In the case of an RPC request, the sink is required.
-    ///
-    /// # Arguments
-    ///
-    /// * `attributes` - `UAttributes` object containing the sink to validate.
+    /// Verifies that attributes for a message representing an RPC request indicate the method to invoke.
     ///
-    /// # Returns
+    /// # Errors
     ///
-    /// Returns a `ValidationResult` that is success or failed with a failure message.
+    /// If the [`UAttributes::sink`] property does not contain a URI representing a method according to
+    /// [`UriValidator::validate_rpc_method`], an error is returned.
     fn validate_sink(&self, attributes: &UAttributes) -> Result<(), UAttributesError> {
         if let Some(sink) = attributes.sink.as_ref() {
             UriValidator::validate_rpc_method(sink)
                 .map_err(|e| UAttributesError::validation_error(e.to_string()))
         } else {
-            Err(UAttributesError::validation_error("Missing Sink"))
+            Err(UAttributesError::validation_error("Attributes for a request message must contain a method-to-invoke in the sink property"))
         }
     }
 
-    /// Validate the time to live configuration. In the case of an RPC request,
-    /// the time to live is required.
+    /// Verifies that a set of attributes representing an RPC request contain a valid time-to-live.
     ///
-    /// # Arguments
+    /// # Errors
     ///
-    /// * `attributes` - `UAttributes` object containing the ttl to validate.
+    /// Returns an error if [`UAttributes::ttl`] (time-to-live) is empty or contains a value less than 1.
+    fn validate_ttl(&self, attributes: &UAttributes) -> Result<(), UAttributesError> {
+        match attributes.ttl {
+            Some(ttl) if ttl > 0 => Ok(()),
```

Comment:
Looks like to me ttl of 0 is still valid, it just means "don't timeout".

[Reference here](https://github.com/eclipse-uprotocol/up-core-api/blob/dc353878914b3459edf2e04531e9cadfc752a2e9/uprotocol/uattributes.proto#L64-L68)

Reply:
A request message actually MUST have a TTL > 0

https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/cloudevents.adoc#21-common-attributes

This seems to be inconsistent with the definition of [CallOptions](https://github.com/eclipse-uprotocol/up-core-api/blob/main/uprotocol/uattributes.proto), though (sigh).

Reply:
I felt that sigh through my screen hah.

So does that mean we'd need two different behaviors or to reconcile this discrepancy with CallOptions?

Reply:
FMPOV it makes sense to require a request to always time-out after a certain time. Publish messages, on the other hand, might *live forever*, if so desired ...
So yes, I guess we need two different checks, depending on the type of message.

Reply:
> FMPOV it makes sense to require a request to always time-out after a certain time. Publish messages, on the other hand, might _live forever_, if so desired ...

Yah, it'd be weird to be like hey here's my request, don't bother getting back to me :sweat_smile: 

In this case should we open an issue on `up-core-api` to clarify that the ttl of 0 should only be valid for `Publish`, `Notification` and `Response`, but not for `Request`?

Better that we align on this I guess at the higher level of spec / core api IMHO.

WDYT?

Reply:
I am currently preparing a PR for https://github.com/eclipse-uprotocol/up-spec/issues/71 which will include this as well ...

Reply:
Just making sure on this one -- did you intend to change this now
```rust
Some(ttl) if ttl > 0 => Ok(()),
```
to align with what we chatted about up-thread or wait on the spec change and do in a follow-up?

Reply:
I was not aware of any changes that are necessary here? Can you elaborate?

Reply:
D'oh! Missed that this was within the impl for Requests and Publish will use the default impl of `validate_ttl()`, which doesn't put a requirement on ttl to be non-zero.